### PR TITLE
Caff 7 - IGVC worlds

### PR DESCRIPTION
**Pull request by 1999michael**
To test, run `roslaunch description caffeine_gazebo.launch`. You can also add additional cmd line parameters to the roslaunch call where:
- 'spawn:= start' (default) or 'spawn:=ramp' to spawn at the ramp for testing
- 'world:= walls' (default) or 'world:=ramp' or 'world:=plain'

### CODE REVIEW

File: worlds/README.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+ - `./install_models.sh`
+2. Build src
+ - `cd ~/SLAM/slam-ws`
+ - `catkin_make`
+
+### Possible Actions
+1. Launch only the world in gazebo
+ - `gazebo ~/SLAM/slam-ws/src/igvc_world/worlds/igvc_basic.world`
+2. Launch the world and caffeine
+ - `roslaunch caffeine caffeine_gazebo.launch world:="ABSOLUTE_PATH_TO_WORLD_FILE"`
```
Comment (erickmu1) [inline worlds/README.md:15]: We may want to update this readme for this specific repository (has the instructions for the slam repo)

Comment (1999michael) [reply worlds/README.md:15]: lmao forgot bout that

File: robot/description/launch/caffeine_gazebo.launch [hunk 2]
```diff
@@ -1,19 +1,33 @@
...
+   <arg name="roll"  value="0"      if="$(eval arg('spawn') == 'start')"/>
+   <arg name="pitch" value="0"      if="$(eval arg('spawn') == 'start')"/>
+   <arg name="yaw"   value="3.14"   if="$(eval arg('spawn') == 'start')"/>
+
+   <!-- Spawn at ramp -->
+   <arg name="x"     value="-19.4"  if="$(eval arg('spawn') == 'ramp')"/>
+   <arg name="y"     value="-12.7"  if="$(eval arg('spawn') == 'ramp')"/>
+   <arg name="z"     value="0.24"   if="$(eval arg('spawn') == 'ramp')"/>
+   <arg name="roll"  value="0"      if="$(eval arg('spawn') == 'ramp')"/>
+   <arg name="pitch" value="0"      if="$(eval arg('spawn') == 'ramp')"/>
```
Comment (erickmu1) [inline robot/description/launch/caffeine_gazebo.launch:25]: We can use a `<group>` tag to minimize the number of if-statments

File: robot/description/launch/caffeine_gazebo.launch [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
-   <arg name="pitch" value="0"/>
-   <arg name="yaw" value="0"/>
-   
+   <arg name="world" default="walls"/>
+   <arg name="spawn" default="start"/>
+
+   <!-- IGVC worlds -->
+   <arg name="world_file" value="$(find worlds)/gazebo_worlds/igvc_walls.world"  if="$(eval arg('world') == 'walls')"/>
+   <arg name="world_file" value="$(find worlds)/gazebo_worlds/igvc_ramp.world"   if="$(eval arg('world') == 'ramp')"/>
+   <arg name="world_file" value="$(find worlds)/gazebo_worlds/igvc_plain.world"  if="$(eval arg('world') == 'plain')"/>
```
Comment (erickmu1) [inline robot/description/launch/caffeine_gazebo.launch:10]: given these if statements, maybe we want to launch the gazebo world in a separate launch file, which we call from this one?

File: robot/description/launch/caffeine_gazebo.launch (hunk 2 above)
Comment (1999michael) [reply robot/description/launch/caffeine_gazebo.launch:25]: actually tried group tag first, but ros doesn't let you define arg values more than once within a launch file (unless you have some if/unless statement)

File: robot/description/launch/caffeine_gazebo.launch (hunk 3 above)
Comment (1999michael) [reply robot/description/launch/caffeine_gazebo.launch:10]: I agree this edit would be cleaner (will implement)

**Pull request merged**